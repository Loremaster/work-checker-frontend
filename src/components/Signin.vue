<template>
  <v-card class="elevation-12">
    <v-toolbar dark color="primary">
      <v-toolbar-title>Login form</v-toolbar-title>
      <v-spacer />
    </v-toolbar>
    <v-card-text>
      <v-form>
        <v-text-field prepend-icon="person" name="email" label="Email" type="email" v-model="email" />
        <v-text-field prepend-icon="lock" name="password" label="Password" type="password" v-model="password" />
      </v-form>
    </v-card-text>
    <v-card-actions>
      <v-spacer />
      <v-btn color="primary" class="submit" @click="login" :disabled="disabled">Login</v-btn>
    </v-card-actions>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import { Action } from "vuex-class";

@Component
export default class Signin extends Vue {
  email: string = "";
  password: string = "";
  disabled: boolean = false;

  @Action("authenticate") authenticate: any;

  async login() {
    try {
      this.disabled = true;
      await this.authenticate({ email: this.email, password: this.password });
      this.$router.push("/");
    } catch (e) {
      this.$notify({
        type: "error",
        title: "Authentication",
        text: "Can't sign in. Please make sure your login and password are correct",
      });
    } finally {
      this.disabled = false;
    }
  }
}
</script>
