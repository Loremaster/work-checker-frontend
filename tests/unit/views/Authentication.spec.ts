import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vue from "vue";
import Component from "@/views/Authentication.vue";
import Signin from "@/components/Signin.vue";

describe("Authentication.vue", () => {
  const setup = () => {
    return shallowMount(Component);
  };

  it("renders correctly", () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Signin component", () => {
    const wrapper = setup();
    const signin = wrapper.find(Signin);

    expect(signin.is(Signin)).toEqual(true);
  });
});
