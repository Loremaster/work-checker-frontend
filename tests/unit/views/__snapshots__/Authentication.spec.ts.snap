// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication.vue renders correctly 1`] = `
<v-container-stub tag="div" fluid="" fill-height="">
  <v-layout-stub tag="div" align-center="" justify-center="">
    <v-flex-stub tag="div" xs12="" sm8="" md4="">
      <signin-stub></signin-stub>
    </v-flex-stub>
  </v-layout-stub>
</v-container-stub>
`;
